src/p2p.rs

```diff
@@ -93,19 +93,31 @@ impl Connection {
             .collect();
         debug!("loading {} blocks", blockhashes.len());
         self.send(NetworkMessage::GetData(inv))?;
-        for hash in blockhashes {
-            match self
-                .recv()
-                .with_context(|| format!("failed to get block {}", hash))?
-            {
-                NetworkMessage::Block(block) => {
-                    assert_eq!(block.block_hash(), hash, "got unexpected block");
+
+        rayon::scope(|s| {
+            let (tx, rx) = crossbeam_channel::bounded(1);
```

Comment:
Perhaps make it a bit larger (10 or something) to even-out any glitches in processing speed?